app/models/maintenance_tasks/parameters/type_error.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+module MaintenanceTasks
+  module Parameters
+    # Error to raise when parameter ActiveModel::Value::Type objects fail
+    # validation.
+    class TypeError < StandardError; end
```

Comment:
This means types implemented by users would need to use this to support error handling. They probably don't want to couple their types with the gem though.